lib/phoenix_html/link.ex

```diff
@@ -46,20 +46,22 @@ defmodule Phoenix.HTML.Link do
 
   def link(text, opts) do
     {to, opts} = Keyword.pop(opts, :to)
-    {method, opts} = Keyword.pop(opts, :method, :get)
-
-    unless to do
+  rescue
+    _ -> raise ArgumentError, "link/2 requires a keyword list as second argument"
```

Comment:
Let's not rescue if we don't need to as it can hide other stuff. Instead, let's check explicitly.